rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -924,6 +924,73 @@ void saveFederationElection_shouldSaveInStorage() {
     }
 
+    @Nested
+    @Tag("proposed federation tests")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class ProposedFederationTests {
+        private final ActivationConfig.ForBlock preLovellActivations = ActivationConfigsForTest.arrowhead631().forBlock(0L);
+        private final ActivationConfig.ForBlock lovellActivations = ActivationConfigsForTest.lovell700().forBlock(0L);
+        private final Federation proposedFederation = new P2shErpFederationBuilder().build();
+        private StorageAccessor bridgeStorageAccessor;
+        private FederationStorageProvider federationStorageProvider;
+
+        @BeforeEach
+        void setUp() {
+            bridgeStorageAccessor = new InMemoryStorage();
+            federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+        }
+
+        @Test
+        void saveProposedFederation_whenProposedFederationIsNotSet_shouldNotSave() {
+            federationStorageProvider.save(networkParameters, lovellActivations);
+
+            assertNull(getProposedFederationFromRepository());
+        }
+
+        @Test
+        void saveProposedFederation_preRSKIP419_whenProposedFederationIsSet_shouldNotSave() {
+            federationStorageProvider.setProposedFederation(proposedFederation);
+            federationStorageProvider.save(networkParameters, preLovellActivations);
+
+            assertNull(getProposedFederationFromRepository());
+        }
+
+        @Test
+        void saveProposedFederation_whenProposedFederationIsSet_shouldSave() {
+            federationStorageProvider.setProposedFederation(proposedFederation);
+            federationStorageProvider.save(networkParameters, lovellActivations);
+
+            assertEquals(proposedFederation, getProposedFederationFromRepository());
+        }
+
+        @Test
+        void saveProposedFederation_whenProposedFederationIsSetToNull_shouldSave() {
+            // first save a non-null value to make sure saving a null one is actually happening
+            federationStorageProvider.setProposedFederation(proposedFederation);
+            federationStorageProvider.save(networkParameters, lovellActivations);
+
+            federationStorageProvider.setProposedFederation(null);
+            federationStorageProvider.save(networkParameters, lovellActivations);
+
+            assertNull(getProposedFederationFromRepository());
+        }
+
+        private Federation getProposedFederationFromRepository() {
+            return bridgeStorageAccessor.getFromRepository(
+                PROPOSED_FEDERATION.getKey(),
+                data -> {
+                    if (data == null) {
+                        return null;
+                    }
+
+                    // storage version should be always present for non-null proposed federation
+                    Integer storageVersion = bridgeStorageAccessor.getFromRepository(PROPOSED_FEDERATION_FORMAT_VERSION.getKey(), BridgeSerializationUtils::deserializeInteger);
```

Comment:
Should we assert this?

Reply:
same as below. Is covered with the existing tests since the storage version is needed to deserialize the fed. But I can add specific tests for this, of course :)